### PR TITLE
Add fail Presto function

### PR DESCRIPTION
**Pull request by rrando901**
Delivers:

- https://github.com/prestodb/presto/issues/21240
- https://github.com/prestodb/presto/issues/21187
- https://github.com/prestodb/presto/issues/20742
- https://github.com/prestodb/presto/issues/21194

### CODE REVIEW

**Comment by czentgr** [comment]
@rrando901 please fix the format errors.

File: velox/functions/prestosql/registration/GeneralFunctionsRegistration.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include "velox/functions/prestosql/Cardinality.h"
 #include "velox/functions/prestosql/GreatestLeast.h"
 #include "velox/functions/prestosql/InPredicate.h"
+#include "velox/functions/prestosql/FailureFunction.h"
```
Comment (czentgr) [inline velox/functions/prestosql/registration/GeneralFunctionsRegistration.cpp:22]: Please put the new include in the correct alphabetical order.

**Review by czentgr — COMMENTED** [review]
Thank you for the change. A few comments.

**Review by czentgr — COMMENTED** [review]
Thanks for this PR!
Can we add a unit test? I thought we had one before?
Also what happens if this is run through the function fuzzer?

**Comment by amitkdutta** [comment]
@rrando901 Are you still working on it? I have similar comments as @czentgr. I can proceed with it if you are not actively working on this PR. Thanks.
CC: @mbasmanova

**Comment by rrando901** [comment]
@amitkdutta Yes, I am actively working on this. Will be pushing up changes to add unit testing soon.

**Comment by rrando901** [comment]
My apologies @amitkdutta. Are you still able to pick this one up? I've been pulled into some other work that requires my attention.

**Comment by mbasmanova** [comment]
Superseded by #10191

**Pull request closed without merging**